kubernetes/launch-match.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash -e
+
+####################
+# Argument parsing #
+####################
+
+DEFAULT_NUM_GPUS=1
+
+usage() {
+  echo "Schedules a job that runs \`match\`."
+  echo
+  echo "Usage: $0 [--gpus GPUS] [--games NUM_GAMES] [--use-weka] PREFIX VICTIM ADVERSARY"
+  echo "          [--EXTRA_MATCH_FLAGS]"
+  echo
+  echo "positional arguments:"
+  echo "  PREFIX     Identifying label used for the name of the job and the name"
+  echo "             of the output directory."
+  echo "  VICTIM     Filename of the victim checkpoint inside the container."
+  echo "  ADVERSARY  Filename of the adversary checkpoint inside the container."
+  echo
+  echo "optional arguments:"
+  echo "  -g GPUS, --gpus GPUS"
+  echo "    Number of GPUs to use."
+  echo "    default: ${DEFAULT_NUM_GPUS}"
+  echo "  -n NUM_GAMES, --games NUM_GAMES"
+  echo "    Number of match games to play. If not specified, then the number of"
+  echo "    games will be the numGamesTotal specified in the \`match\` config"
+  echo "    multiplied by the number of GPUs."
+  echo "  -w, --use-weka"
+  echo "    Store results on the go-attack Weka volume instead of the CHAI NAS"
+  echo "    volume."
+  echo
+  echo "Optional arguments should be specified before positional arguments."
+  echo
+  echo "Extra flags to \`match\` can be specified by adding them after the"
+  echo "positional arguments with \"--\" in between, e.g.,"
+  echo "  $0 test-run -- -override-config nnModelFile0=/dev/null"
+}
+
+NUM_POSITIONAL_ARGUMENTS=3
+
+NUM_GPUS=${DEFAULT_NUM_GPUS}
+# Command line flag parsing (https://stackoverflow.com/a/33826763/4865149)
+while true; do
+  case $1 in
+    -h|--help) usage; exit 0 ;;
+    -g|--gpus) NUM_GPUS=$2; shift ;;
+    -n|--games) NUM_GAMES=$2; shift ;;
+    -w|--use-weka) USE_WEKA=1 ;;
+    *) break ;;
+  esac
+  shift
+done
+
+if [ $# -lt ${NUM_POSITIONAL_ARGUMENTS} ]; then
+  usage
+  exit 1
+fi
+
+PREFIX=$1
+VICTIM=$2
+ADVERSARY=$3
+RUN_NAME="$PREFIX-$(date +%Y%m%d-%H%M%S)"
+echo "Run name: $RUN_NAME"
+shift 3
+
+if [ $# -gt 0 ]; then
+  if [ "$1" != "--" ]; then
+    echo "Unexpected arguments: $*"
+    usage
+    exit 1
+  fi
+  shift
+  # remaining arguments are treated as extra arguments to `match`
+fi
+
+############################
+# Launching the experiment #
+############################
+
+# shellcheck disable=SC1091
+source "$(dirname "$(readlink -f "$0")")"/launch-common.sh
+
+if [ -n "${USE_WEKA}" ]; then
```

Comment:
Lines 84-88 are duplicated with `launch-job.sh`, considering factoring them out into a bash function ala https://tldp.org/HOWTO/Bash-Prog-Intro-HOWTO-8.html

I think this function can set VOLUME_FLAGS and access USE_WEKA directly because bash everything is global namespace by default...

So you could do something like `set_volume_flags()` call from here and have:

```
set_volume_flags() {
    if [ -n "${USE_WEKA}" ]; then
      # ...
```